### PR TITLE
Use LLMChain for base prompt llm invocation

### PR DESCRIPTION
**Pull request by bparees**
## Description

LLM loader returns a `ChatOpenAI` instance(derived from `BaseChatModel`) for openai, whereas for BAM the loader returns a `LangChainInterface` (derived from `LLM`), and those two base classes don't have the same invocation arguments(one takes the prompt in the form of a raw string, the other expects a List of BaseMessages), 

This meant we weren't able to use the result of llmloader directly(it would fail when switching to openai, the code worked for the LLM base class but not for the BaseChatModel class).  

All the rest of our code seems to wrap the result of llmloader into a `LLMChain` which seems to abstract the argument difference, so i've updated the base_prompt implementation to do the same.....but this is going to be something to keep an eye on (using LLM vs BaseChatModel interfaces in OLS, and whether there are other differences under the covers even when wrapping them in a LLMChain, or other behaviors of LLMChain that we do not necessarily want)

## Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: tests/integration/test_ols.py [hunk 1]
```diff
@@ -45,19 +47,10 @@ def test_status() -> None:
     }
 
 
+# the raw prompt should just return stuff from LLMChain, so mock that base method in ols.py
+@patch("ols.app.endpoints.ols.LLMChain", new=mock_llm_chain("test response"))
 def test_raw_prompt(monkeypatch) -> None:
```
Comment (onmete) [inline tests/integration/test_ols.py:52]: You can remove `monkeypatch` here. It's not used anymore.

**Comment by thoraxe** [comment]
/approve

but do we still want this endpoint? It was mostly a hack/test

File: tests/integration/test_ols.py (hunk 1 above)
Comment (bparees) [reply tests/integration/test_ols.py:52]: good catch, we'll get it cleaned up in a follow up

**Pull request merged**